sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -403,55 +409,26 @@ class SymmetricHashJoinStateManager(
   def getInternalRowOfKeyWithIndex(currentKey: UnsafeRow): InternalRow = keyProjection(currentKey)
 
   /** Commit all the changes to all the state stores */
-  def commit(): Unit = {
-    keyToNumValues.commit()
-    keyWithIndexToValue.commit()
-  }
+  def commit(): Unit
 
   /** Abort any changes to the state stores if needed */
-  def abortIfNeeded(): Unit = {
-    keyToNumValues.abortIfNeeded()
-    keyWithIndexToValue.abortIfNeeded()
-  }
+  def abortIfNeeded(): Unit
 
   /**
    * Get state store checkpoint information of the two state stores for this joiner, after
    * they finished data processing.
+   *
+   * For [[SymmetricHashJoinStateManagerV1]], this returns the information of the two stores
```

Comment:
nit: Let's document this in each derived class. As long as the behavior fits to the base class's contract, it should be good.